### PR TITLE
Feature/add content model

### PR DESCRIPTION
**Pull request by sudan45**
Addresses
- https://github.com/toggle-corp/ai-chatbot-backend/issues/15

NOTE: **Mention related users here if any.**

## Changes

* Detailed list or prose of changes
* Breaking changes
* Changes to configurations

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)

### CODE REVIEW

File: content/admin.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.contrib import admin
+
+from content.models import Content
+
+# Register your models here.
+
+
+@admin.register(Content)
+class ContentAdmin(admin.ModelAdmin):
+    list_display = ["title", "content_id"]
```
Comment (thenav56) [inline content/admin.py:17]: Missing auto complete fields

File: content/models.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    title = models.CharField(max_length=100)
+    document_type = models.IntegerField(choices=DocumentType.choices)
+    document_file = models.FileField(upload_to="documents", blank=True)
+    extracted_file = models.FileField(upload_to="documents", blank=True)
+    content_id = models.UUIDField(default=uuid.uuid4, editable=False)
+    description = models.TextField(help_text="Content text")
+    tag = models.ManyToManyField("Tag", blank=True)
+    is_deleted = models.BooleanField(default=False)
+    deleted_at = models.DateTimeField(blank=True)
```
Comment (thenav56) [inline content/models.py]: `null=True` is also required. `blank` is only for forms

File: content/models.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    class DocumenetStatus(models.IntegerChoices):
+        PENDING = 1, _("Pending")
+        TEXT_EXTRACTED = 2, _("Text extracted")
+        ADDED_TO_VECTOR = 3, _("Added to vector")
+        DELETED_FROM_VECTOR = 4, _("Deleted from vector")
+
+    title = models.CharField(max_length=100)
+    document_type = models.IntegerField(choices=DocumentType.choices)
+    document_file = models.FileField(upload_to="documents", blank=True)
+    extracted_file = models.FileField(upload_to="documents", blank=True)
```
Comment (thenav56) [inline content/models.py]:
```suggestion
    extracted_file = models.FileField(upload_to="documents-extracts", blank=True, null=True)
```

File: content/models.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    class DocumenetStatus(models.IntegerChoices):
+        PENDING = 1, _("Pending")
+        TEXT_EXTRACTED = 2, _("Text extracted")
+        ADDED_TO_VECTOR = 3, _("Added to vector")
+        DELETED_FROM_VECTOR = 4, _("Deleted from vector")
+
+    title = models.CharField(max_length=100)
+    document_type = models.IntegerField(choices=DocumentType.choices)
+    document_file = models.FileField(upload_to="documents", blank=True)
```
Comment (thenav56) [inline content/models.py]: Any reason why this is `blank=True`?

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: content/models.py (hunk 4 above)
Comment (sudan45) [reply content/models.py]: for this phase we accept content text only

File: content/models.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        ADDED_TO_VECTOR = 3, _("Added to vector")
+        DELETED_FROM_VECTOR = 4, _("Deleted from vector")
+        FAILURE = 5, _("Failure")
+
+    title = models.CharField(max_length=100)
+    document_type = models.IntegerField(choices=DocumentType.choices, default=DocumentType.WORD)
+    document_file = models.FileField(upload_to="documents", blank=True)
+    extracted_file = models.FileField(upload_to="documents-extracts", null=True, blank=True)
+    content_id = models.UUIDField(default=uuid.uuid4, editable=False)
+    description = models.TextField(help_text="Content text")
```
Comment (thenav56) [inline content/models.py]: `description` Remove this.

File: content/models.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class Tag(models.Model):
+    name = models.CharField(max_length=20)
+    description = models.CharField(max_length=50, null=True, blank=True)
+
+
+class Content(UserResource):
+    class DocumentType(models.IntegerChoices):
+        WORD = 1, _("Word")
+        PDF = 2, _("PDF")
```
Comment (thenav56) [inline content/models.py:20]: Add TEXT

File: content/models.py (hunk 4 above)
Comment (thenav56) [reply content/models.py]: This should be required, if type is TEXT, use the same path for extracted_file

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: content/models.py [hunk 7]
```diff
@@ -24,13 +28,21 @@ class DocumentStatus(models.IntegerChoices):
...
     deleted_by = models.ForeignKey("user.User", null=True, blank=True, on_delete=models.PROTECT)
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
+        """Save the content to the database."""
+        if self.document_type == self.DocumentType.TEXT:
+            self.extracted_file = self.document_file
+        super().save(*args, **kwargs)
```
Comment (thenav56) [inline content/models.py]: Only do this if self.pk is None

**Review by thenav56 — APPROVED** [review]

**Pull request merged**